scripts/build_osm_network.py

```diff
@@ -719,10 +719,13 @@ def fix_overpassing_lines(lines, buses, distance_crs, tol=1):
     # remove original lines
     lines.drop(lines_to_split, inplace=True)
 
-    lines = gpd.GeoDataFrame(
-        pd.concat([lines, df_to_add], ignore_index=True).reset_index(drop=True),
-        crs=lines.crs,
-    )
+    if lines.empty:
+        lines = gpd.GeoDataFrame(df_to_add, crs=lines.crs)
```

Comment:
Not sure about the index: it may be the case that it's safer to reset it, as it's done for the concatenated data frame bellow. It would be great to test.

Reply:
Hi @ekatef , thanks :)

 I am not sure however how to perform this test. I thought that such test was included in our regression tests. How do you reckon we should proceed?

Reply:
Unfortunately, not. CI tests include rather standard cases (more or less corresponding to the tutorial case which deals with Nigeria + Benin). If there is need to test more sophisticated cases (for example, if the PR may behave differently for different grid topologies like in our case), some additional test cases may be needed.

Not sure however for which countries can we expect to have all the lines to be splited... @davide-f by any chance, are you aware on the region, where all the original lines are removed by this fragment: https://github.com/pypsa-meets-earth/pypsa-earth/blob/cea0dc03bf6e718504eada9dbb1eff9a844f18f2/scripts/build_osm_network.py#L719-L720
?

Reply:
If lines is empty this entire function looses meaning:
This function aims to split line fragments where there is a substation on its path.
So this exception should never occur.

Reply:
> If lines is empty this entire function looses meaning: This function aims to split line fragments where there is a substation on its path. So this exception should never occur.

@davide-f  thanks for looking into that! :)
I mean rather a situation when all lines should be split which means that the line `lines.drop(lines_to_split, inplace=True)` removes all the lines from the network. Is it not possible?

The point is, `pandas` complained about concatenating dataframes, while one of them is empty or all NAs. Looking into this check,

https://github.com/pypsa-meets-earth/pypsa-earth/blob/cea0dc03bf6e718504eada9dbb1eff9a844f18f2/scripts/build_osm_network.py#L706-L707

I have an impression that `lines_to_add` can't be empty, so the deprecation problem arises if there are no lines left in `n.lines`, after removing `lines_to_split`. Am I missing something?

Reply:
Ahhh I get it. Actually I don't know, it may be an edge case of small countries with very few lines.
The previous code never crashed though

Reply:
Hi all,
I modified the code with 

    lines = df_to_add if lines.empty else pd.concat([lines, df_to_add])

    lines = gpd.GeoDataFrame(lines.reset_index(drop=True), crs=lines.crs)

In this way the pipelines are still working and deprecation warning disappears.

Reply:
Perfect!! 😄